Classes/Domain/TemplateNodeCreationHandler.php

```diff
@@ -46,7 +46,9 @@ public function handle(NodeInterface $node, array $data): void
 
         $evaluationContext = [
             'data' => $data,
+            // deprectated will be removed in 3.0
             'triggeringNode' => $node,
+            'parentNode' => $node->getParent()
```

Comment:
@Sebobo and me are still unsure about the naming, as its not clear it always means the same

Reply:
We need a materialized node that can be used in flow queries for example. What about  "referenceNode" or "existingNode"? On the other hand, if it really always is the parent node of the "triggering node" we provide, why not simply call it that? I know, the behaviour has to change (it's not the parent node of the current node in context any more). However, I'd prefer to have a clear name for the new version, even if it may cause some confusion for someone used to the old version. WDYT?

Reply:
parentNode still sounds best I agree, but referenceNode would be better documentation wise and would lead to less accidental confusions depending on the context.

Reply:
I think referenceNode is ambiguous as one might confuse it with the „references“ feature.

Maybe `contextNode` but I dislike the word context - it’s too overused

Reply:
Context variables by default cause confusion. I think with `referenceNode` we do the least harm.

Reply:
Maybe `insertionReferenceNode` or `ankhorNode`

Reply:
how about `site` for the current siteNode?

Reply:
hmm how about just simply `sourceNode` and `site`?

Reply:
I dont like `referenceNode` or the like because `reference` will let people think of node references

Reply:
or for our usecase @Sebobo the current `documentNode` would suffice as well ...

Reply:
`triggeringNode` is currently the to be created node but in Neos9 it will  be deprecated and null
`parentNode` will be the parent node independent of the current insertion anchor thingy
`site` will be the site node


Depending on the current node / select node is not something we want to expose as its fragile and depends what the editor did and is nothing predictable.